graylog2-server/src/main/java/org/graylog2/indexer/rotation/strategies/TimeBasedSizeOptimizingStrategy.java

```diff
@@ -100,21 +105,25 @@ protected Result shouldRotate(final String index, IndexSet indexSet) {
     }
 
     private boolean indexExceedsLeeWay(DateTime creationDate, Period leeWay) {
-        final Instant now = Instant.now();
+        final Instant now = clock.instantNow();
         final Instant leewayLimit = now.minus(ROTATION_PERIOD.plus(leeWay));
 
-        return creationDate.isBefore(leewayLimit.toEpochMilli());
+        final Duration between = Duration.between(Instant.ofEpochMilli(creationDate.getMillis()), now);
+        return between.compareTo(Duration.ofDays(ROTATION_PERIOD.plus(leeWay).getDays())) >= 0;
+        //return creationDate.isBefore(leewayLimit.toEpochMilli());
     }
 
     private boolean indexIsOldEnough(DateTime creationDate) {
-        final Instant now = Instant.now();
+        final Instant now = clock.instantNow();
         final Instant rotationLimit = now.minus(ROTATION_PERIOD);
 
-        return creationDate.isBefore(rotationLimit.toEpochMilli());
+        final Duration between = Duration.between(Instant.ofEpochMilli(creationDate.getMillis()), now);
+        return between.compareTo(Duration.ofDays(ROTATION_PERIOD.getDays())) >= 0;
+        //return creationDate.isBefore(rotationLimit.toEpochMilli());
     }
 
     private boolean indexExceedsSizeLimit(long size) {
-        return size > MAX_INDEX_SIZE.toBytes();
+        return size >= MAX_INDEX_SIZE.toBytes();
```

Comment:
The name `indexExceedsSizeLimit` is a bit misleading, if we allow equality. Also the log messages that rely on this method explicitly say "exceeded".
1 byte up or down is insignificant for what is typically measured in GB. Why is this change needed?

Reply:
to accommodate the test. but yeah, I can revert that